replication/event.go

```diff
@@ -356,7 +356,7 @@ func (e *QueryEvent) Dump(w io.Writer) {
 	fmt.Fprintln(w)
 }
 
-type GTIDEvent struct {
+type MySQLGTIDEvent struct {
```

Comment:
I guess you rename this struct to avoid using the same name with the new interface `GTIDEvent`? But it will break more compatibility. I'll think if we have a better way to handle it later

Reply:
I could call it GenericGTIDEvent or something?

Wouldn't be what we would probably name it originally, but would break less compatibility.

Reply:
I went with BinlogGTIDEvent. Let me know what you think.

Reply:
lgtm for the name. I'm a bit busy these days, will check the whole PR later